src/hooks/useAdvanceStep.js

```diff
@@ -0,0 +1,34 @@
+import { useRef, useState } from 'react'
+
+function useAdvanceStep() {
+  const ref = useRef(null)
+
+  const [currentStep, setCurrentStep] = useState(0)
+
+  const scrollTop = () => window.scrollTo({ x: 0 })
+
+  const changeTitle = title => (document.title = title)
+
+  const setFocus = () => {
+    setTimeout(() => {
+      ref.current.setAttribute('tabindex', '-1')
+      ref.current.focus()
+      ref.current.removeAttribute('tabindex')
+    }, 0)
+  }
```

Comment:
has dependecy on ref but since ref is constant can be wrapped in useCallback to prevent unnecessary redefinitions.